### PR TITLE
Eigen alignment fix

### PR DESCRIPTION
**Pull request by akirayou**
I fixed Eigen Alignment problem .
Now It can run on Ubuntu16 / Eigen 3.3/ROS(kinetic).

### CODE REVIEW

File: src/util/NumType.h [hunk 1]
```diff
@@ -27,10 +27,12 @@
 #include "Eigen/Core"
 #include "sophus/sim3.hpp"
 #include "sophus/se3.hpp"
-
+#include <vector>
 
 namespace dso
 {
+template <class T, class Allocator =  Eigen::aligned_allocator<std::pair<const int,T>>   >
+using vectorA = std::vector< T, Allocator > ;
```
Comment (pmoulon) [inline src/util/NumType.h]: Same comment as https://github.com/uzh-rpg/rpg_svo/pull/193#discussion_r108003101
Using
```
#include<Eigen/StdVector>
EIGEN_DEFINE_STL_VECTOR_SPECIALIZATION(VecX)
```
will be less intrusive.

**Comment by akirayou** [comment]
And if you want use this with dso_ros. you have to modify CXX_FLAGS in CMakeList.txt.    
dso and dso_ros must have same CXX_FLAGS ,especially "-march=native" flag. -march and -sse flags changes Eigen's alignment size.

**Comment by akirayou** [comment]
I forgot that check VecX is dynamic size. Dynamic size vector do not need alignment allocator.
So, I remove them.

**Comment by JakobEngel** [comment]
merged! thanks.

**Comment by a-jahani** [comment]
After merging dso_ROS will give us error in compiling. 
SO I tried to just add one line to coarseinitializer.h 
`#define EIGEN_ALIGN32 EIGEN_ALIGN_TO_BOUNDARY(32)`
and I tested with eigen 3.2 and 3.3.3 and both were OK. I think you might need to change dso_ROS or go back to previous version

**Pull request closed without merging**